contracts/exchange/contracts/src/MixinMatchOrders.sol

```diff
@@ -33,6 +33,110 @@ contract MixinMatchOrders is
+    )
+        public
+        nonReentrant
+        returns (LibFillResults.BatchMatchedFillResults memory batchMatchedFillResults)
+    {
+        // Ensure that the left and right arrays are compatible and have nonzero lengths.
+        // If these checks fail, revert with accurate RichRevert reasons.
+        if (leftOrders.length == 0) {
+            _rrevert(BatchMatchOrdersError(BatchMatchOrdersErrorCodes.ZERO_LEFT_ORDERS));
+        }
+        if (rightOrders.length == 0) {
+            _rrevert(BatchMatchOrdersError(BatchMatchOrdersErrorCodes.ZERO_RIGHT_ORDERS));
+        }
+        if (leftOrders.length != leftSignatures.length) {
+            _rrevert(BatchMatchOrdersError(BatchMatchOrdersErrorCodes.INCOMPATIBLE_LEFT_ORDERS));
+        }
+        if (rightOrders.length != rightSignatures.length) {
+            _rrevert(BatchMatchOrdersError(BatchMatchOrdersErrorCodes.INCOMPATIBLE_RIGHT_ORDERS));
+        }
+
+        // Without simulating all of the order matching, this program cannot know how many
+        // matches there will be. To ensure that batchMatchedFillResults has enough memory
+        // allocated for the left and the right side, we will allocate enough space for the
+        // maximum amount of matches (the maximum of the left and the right sides).
+        uint256 maxLength = _max256(leftOrders.length, rightOrders.length);
+        batchMatchedFillResults.left = new LibFillResults.FillResults[](maxLength);
+        batchMatchedFillResults.right = new LibFillResults.FillResults[](maxLength);
+
+        // Initialize initial variables
+        uint256 matchCount;
+        uint256 leftIdx = 0;
+        uint256 rightIdx = 0;
+        LibOrder.Order memory leftOrder = leftOrders[0];
+        LibOrder.Order memory rightOrder = rightOrders[0];
+        bytes memory leftSignature = leftSignatures[0];
+        bytes memory rightSignature = rightSignatures[0];
+
+        // Loop infinitely (until broken inside of the loop), but keep a counter of how
+        // many orders have been matched.
+        for (matchCount = 0;; matchCount++) {
+            // Match the two orders that are pointed to by the left and right indices
+            LibFillResults.MatchedFillResults memory matchResults = _matchOrders(
+                leftOrder,
+                rightOrder,
+                leftSignature,
+                rightSignature
+            );
+
+            // Add the matchResults and the profit made during the match to the
+            // batchMatchedFillResults for this batch.
+            batchMatchedFillResults.left[matchCount] = matchResults.left;
+            batchMatchedFillResults.right[matchCount] = matchResults.right;
+            batchMatchedFillResults.profitInLeftMakerAsset = _safeAdd(
+                batchMatchedFillResults.profitInLeftMakerAsset,
+                matchResults.leftMakerAssetSpreadAmount
+            );
+            // batchMatchedFillResults.profitInRightMakerAsset += 0; // Placeholder for ZEIP 40
```

Comment:
I know I told you to just comment this out before, but seeing as the ZEIP-40 PR is up now, could you retrofit `rightMakerAssetSpreadAmount` into `MatchedFillResults` in this PR and flesh out the logic for this line? I don't want us to forget about this line later.